Source/WebCore/Modules/model-element/HTMLModelElement.cpp

```diff
@@ -688,6 +728,73 @@ void HTMLModelElement::setCurrentTime(double currentTime)
     if (m_modelPlayer)
         m_modelPlayer->setCurrentTime(Seconds(currentTime), [&]() { });
 }
+
+const URL& HTMLModelElement::environmentMap() const
+{
+    return m_environmentMapURL;
+};
+
+void HTMLModelElement::setEnvironmentMap(const URL& url)
+{
+    if (url.string() == m_environmentMapURL.string())
+        return;
+
+    m_environmentMapURL = url;
+
+    m_environmentMapData.reset();
+
+    if (m_environmentMapResource) {
+        m_environmentMapResource->removeClient(*this);
+        m_environmentMapResource = nullptr;
+    }
+
+    if (!m_environmentMapReadyPromise->isFulfilled())
+        m_environmentMapReadyPromise->reject(Exception { ExceptionCode::AbortError });
+
+    m_environmentMapReadyPromise = makeUniqueRef<EnvironmentMapPromise>();
+
+    if (m_environmentMapURL.isEmpty()) {
+        ActiveDOMObject::queueTaskToDispatchEvent(*this, TaskSource::DOMManipulation, Event::create(eventNames().errorEvent, Event::CanBubble::No, Event::IsCancelable::No));
+        return;
```

Comment:
We also need to handle removing the environment map (if it's set from a non-empty value to an empty one, or set from a non-empty IBL to a non-existent one). We'd want to end up calling `removeIBL` on the `WKSRKEntity` in that case.

Reply:
Added code to handle resource removal.